### PR TITLE
feat(app): Add ability to specify tab ordering and custom tab labels

### LINKED ISSUE #522: [FEATURE] Add ability to specify tab ordering and custom tab labels
##### **Overview of the issue**

My team would like to be able to set the order of tabs and customize the tab labels.

##### **Motivation for or Use Case**

We want to be able to make the Examples tab the leftmost tab, and we would also like to set the label for the tab to "Overview" instead of "Examples".

##### **Suggest a Fix**

We would like to contribute this feature. Our plan is to allow the user to optionally pass in an array of tab data that determines the tab order and the tab labels. Although the existing disabling flags would be left in place for consumers who don't want to provide a tabConfig, using this optional tabConfig parameter would obviate the need to disable individual tabs. Additionally, the same parameter could be extended later to allow the consumer to add custom tabs with custom content.

Note: If a consumer specifies a tab that a particular partial template does not currently contain, the tab would be skipped for that partial. For example, if the tabConfig includes a "Template" tab, the Template tab would only appear on the DOM for components and would not appear for classes, directives, injectables, interceptors, interfaces, modules, or pipes to maintain the current behavior.

Example usage: 
```
tabConfig: [
    {
        id: "examples",
        label: "Overview" // customized label
    },
    {
        id: "info",
        label: "API" // customized label
    },
    {
        id: "source",
        label: "Source"
    }
]
```

### PR DESCRIPTION
**Pull request by Blakesalot**
fix #522

### CODE REVIEW

**Comment by vogloblinsky** [comment]
Thanks blake for the PR. I will review it asap while implementing https://github.com/compodoc/compodoc/issues/379

**Comment by Blakesalot** [comment]
Sounds good, Vincent. Thanks.

**Comment by Blakesalot** [comment]
Hi, @vogloblinsky. Not to rush you, but do you have an update on when this PR might be reviewed?

**Comment by vogloblinsky** [comment]
@Blakesalot I will try this week, but i have a conference next thursday in Paris, so may be more next week.

**Comment by Blakesalot** [comment]
Okay. Thank you, Vincent, for the update. :)

File: src/utils/constants.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export const COMPODOC_CONSTANTS = {
  * Max length for the string of a file during Lunr search engine indexing.
  * Prevent stack size exceeded
  */
-export const MAX_SIZE_FILE_SEARCH_INDEX = 15000;
+export const MAX_SIZE_FILE_SEARCH_INDEX = 50000;
```
Comment (Blakesalot) [inline src/utils/constants.ts]: value moved from defaults.ts

**Comment by Blakesalot** [comment]
@vogloblinsky, do you have any updates on when this might be pulled in? Thanks.

**Comment by Blakesalot** [comment]
Sounds great, @vogloblinsky. I've merged develop into my branch along with the config file feature you added for #379. It looks like the custom tabs feature in this PR will need to be integrated into it at around line 223 of index-cli.ts. Will you be doing that?

**Comment by vogloblinsky** [comment]
@Blakesalot Yes i will do that now.

**Comment by vogloblinsky** [comment]
@Blakesalot Could you do a PR for the documentation for this new feature ?

https://github.com/compodoc/website/tree/develop

Thanks

**Comment by Blakesalot** [comment]
Sure. I will confer with my team to see if I can include it within the next couple of sprints.

**Comment by Blakesalot** [comment]
Looks good! I will see about adding a separate page with more info soon. Thanks

**Pull request merged**